Source/ServerTrustEvaluation.swift

```diff
@@ -326,6 +327,137 @@ public final class PublicKeysTrustEvaluator: ServerTrustEvaluating {
+    ///   - performDefaultValidation: Determines whether default validation should be performed in addition to
+    ///                               evaluating the pinned certificates. Defaults to `true`.
+    ///   - validateHost:             Determines whether or not the evaluator should validate the host, in addition to
+    ///                               performing the default evaluation, even if `performDefaultValidation` is `false`.
+    ///                               Defaults to `true`.
+    public init(keys: [String],
+                performDefaultValidation: Bool = true,
+                validateHost: Bool = true) {
+        self.keys = keys
+        self.performDefaultValidation = performDefaultValidation
+        self.validateHost = validateHost
+    }
+
+    private enum Asn1Header {
+        case rsa2048
+        case rsa4096
+        case ecDsaSecp256r1
+        case ecDsaSecp384r1
+
+        var bytes:[UInt8] {
+            switch self {
+            case .rsa2048:
+                return [
+                    0x30, 0x82, 0x01, 0x22, 0x30, 0x0d, 0x06, 0x09, 0x2a, 0x86, 0x48, 0x86,
+                    0xf7, 0x0d, 0x01, 0x01, 0x01, 0x05, 0x00, 0x03, 0x82, 0x01, 0x0f, 0x00
+                ]
+            case .rsa4096:
+                return  [
+                    0x30, 0x82, 0x02, 0x22, 0x30, 0x0d, 0x06, 0x09, 0x2a, 0x86, 0x48, 0x86,
+                    0xf7, 0x0d, 0x01, 0x01, 0x01, 0x05, 0x00, 0x03, 0x82, 0x02, 0x0f, 0x00
+                ]
+            case .ecDsaSecp256r1:
+                return  [
+                    0x30, 0x59, 0x30, 0x13, 0x06, 0x07, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x02,
+                    0x01, 0x06, 0x08, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x03, 0x01, 0x07, 0x03,
+                    0x42, 0x00
+                ]
+            case .ecDsaSecp384r1:
+                return [
+                    0x30, 0x76, 0x30, 0x10, 0x06, 0x07, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x02,
+                    0x01, 0x06, 0x05, 0x2b, 0x81, 0x04, 0x00, 0x22, 0x03, 0x62, 0x00
+                ]
+            }
+        }
+    }
+
+    private func getAsn1HeaderBytes(publicKeyType:String,publicKeySize:Int) -> [UInt8]? {
+        if publicKeyType == kSecAttrKeyTypeRSA as String && publicKeySize == 2048 {
+            return Asn1Header.rsa2048.bytes
+        } else if publicKeyType == kSecAttrKeyTypeRSA as String && publicKeySize == 4096 {
+            return Asn1Header.rsa4096.bytes
+        } else if publicKeyType == kSecAttrKeyTypeECSECPrimeRandom as String && publicKeySize == 256 {
+            return Asn1Header.ecDsaSecp256r1.bytes
+        } else if publicKeyType == kSecAttrKeyTypeECSECPrimeRandom as String && publicKeySize == 384 {
+            return Asn1Header.ecDsaSecp384r1.bytes
+        }
+        return nil // Unknown Algorithm
+    }
+
+    private func hashSubjectPublicKeyInfoFromCertificate(certificate:SecCertificate) -> String? {
```

Comment:
nit: space needed after `certificate:` might also want to rename the function to `subjectPublicKeyHash(from certificate: SecCertificate)` so it reads more swifty